### ISSUE TITLE
MAP casts to invalid JSON

### ISSUE THREAD

**Issue by tekumara** [duckdb/duckdb#6687]
### What happens?

duckdb will cast a MAP to invalid JSON

### To Reproduce

eg:
```
D  SELECT cast(MAP(LIST_VALUE('a', 'b', 'c'), LIST_VALUE(1, 'BBBB', NULL)) as json);
┌───────────────────────────────────────────────────────────────────────────┐
│ CAST(map(list_value('a', 'b', 'c'), list_value(1, 'BBBB', NULL)) AS json) │
│                                   json                                    │
├───────────────────────────────────────────────────────────────────────────┤
│ {a=1, b=BBBB, c=NULL}                                                     │
└───────────────────────────────────────────────────────────────────────────┘
D SELECT json(cast(MAP(LIST_VALUE('a', 'b', 'c'), LIST_VALUE(1, 'BBBB', NULL)) as json));
Error: Invalid Input Error: Malformed JSON at byte 1 of input: unexpected character.  Input: {a=1, b=BBBB, c=NULL}
```

I was expecting the map as json to be something like: `{"a":1, "b":"BBBB"}`

### OS:

macos

### DuckDB Version:

v0.7.1 b00b93f0b1

### DuckDB Client:

duckdb_cli-osx-universal

### Full Name:

Oliver Mannion

### Affiliation:

Xero

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by lnkuiper** [comment]
Thanks for the bug report! For now, this is expected behavior, but this will be changed in the future. Currently, the `to_json` cast will cast the input to a string without creating a valid JSON. You can create valid JSON like so:
```sql
D select to_json(MAP(LIST_VALUE('a', 'b', 'c'), LIST_VALUE(1, 'BBBB', NULL))) j;
┌───────────────────────────────┐
│               j               │
│             json              │
├───────────────────────────────┤
│ {"a":"1","b":"BBBB","c":null} │
└───────────────────────────────┘
```
In the future, I plan to make JSON casts call `to_json` implicitly.

**Comment by lnkuiper** [comment]
Closed via #7366
